### PR TITLE
Response functions

### PR DESCRIPTION
**Pull request by quintaartsen**

### CODE REVIEW

File: src/concretes/peer.ts [hunk 1]
```diff
@@ -3,65 +3,106 @@ import { Guid } from "guid-typescript";
...
+     * Creates an instance of peer.
+     */
+    constructor(type: string) {
         this.connectionService = new ConnectionService();
         this.createReceiverHandlers();
+        this.type = type;
     }
 
+    /**
+     * Inits peer
```
Comment (ronaldvandenbroek) [inline src/concretes/peer.ts:27]: General comment for all instances: Please add explanations to the @params and @returns. The doc doesn't add anything without explanations.

File: src/services/concretes/socketIO/socketIO.service.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import { SocketIOReceiver } from "./socketIOReceiver.service";
+import { SocketIOSender } from "./socketIOSender.service";
+
+/**
+ * Handles the sending of messages.
+ */
+export class SocketIOCommunicationProtocol implements ICommunicationProtocol {
+    private readonly sender: SocketIOSender;
+    private readonly reciever: SocketIOReceiver;
+
```
Comment (hugovandevliert) [inline src/services/concretes/socketIO/socketIO.service.ts]: Please remove this duplicate newline

File: src/models/routingTable.model.ts [hunk 3]
```diff
@@ -1,21 +1,43 @@
...
+    }
+
+
+    /**
+     * Gets peer of type
+     * @param type 
+     * @returns peer of type 
+     */
+    public getPeerOfType(type: string): string | undefined {
+        for (const peer of this.peers) {
```
Comment (ronaldvandenbroek) [inline src/models/routingTable.model.ts]: Why is this peer[1]? Why not use the peer object? Either change or add an explanation in the TSDoc.

File: src/concretes/peer.ts [hunk 4]
```diff
@@ -131,16 +152,55 @@ export class Peer implements IPeer {
             }
             for (const peer of peers) {
                 // Check if peer is online and try to join
-                const message = new Message(MessageType.JOIN, this.connectionService.GUID, JSON.stringify({ip: THIS_IP}));
-                await this.connectionService.sendMessageByIp(message, peer,
-                    async (responseMessage: Message) => { await this.joinResponse(responseMessage); });
+                const message = new Message(MessageType.JOIN, this.connectionService.GUID, JSON.stringify({type: this.type}));
```
Comment (ronaldvandenbroek) [inline src/concretes/peer.ts]: Change the name of 'type', it is not clear if this is referencing to the messageType or peerType.

**Review by ronaldvandenbroek — CHANGES_REQUESTED** [review]

File: src/services/concretes/socketIO/socketIO.service.ts (hunk 2 above)
Comment (timofudala) [reply src/services/concretes/socketIO/socketIO.service.ts]: Please change the `reciever` typo to `receiver`.

**Review by ronaldvandenbroek — COMMENTED** [review]
Why is the pipeline not in use anymore on the last few commits?

**Review by ronaldvandenbroek — APPROVED** [review]

File: src/concretes/peer.ts [hunk 5]
```diff
@@ -3,65 +3,113 @@ import { Guid } from "guid-typescript";
 import { MessageType } from "../enums/messageType.enum";
 import { IPeer, RECIEVE_HANDLER_TYPE, RESPONSE_TYPE } from "../interfaces/peer";
 import { Message } from "../models/message.model";
```
Comment (Juuldamen) [inline src/concretes/peer.ts]: Shorten this import

**Pull request merged**